packages/2018-transportation-systems/src/state/decline-in-ridership/decline-in-ridership.test.js

```diff
@@ -0,0 +1,215 @@
+      data: null,
+    };
+    const payload = { stu: 'ff' };
+
+    it('should return the initial state', () => {
+      expect(reducer(undefined, {})).to.eql(initialState);
+    });
+
+    it('should handle API_START', () => {
+      expect(reducer(initialState, {
+        type: actions.API_START,
+      })).to.eql({
+        pending: true,
+        error: null,
+        data: null,
+      });
+    });
+
+    it('should handle API_SUCCESS', () => {
+      expect(reducer({ pending: true, error: null, data: null }, {
+        type: actions.API_SUCCESS,
+        payload,
+      })).to.eql({
+        pending: false,
+        data: payload,
+      });
+    });
+
+    it('should handle API_ERROR', () => {
+      expect(reducer({ pending: true, error: null, data: null }, {
+        type: actions.API_ERROR,
+        payload,
+      })).to.eql({
+        pending: false,
+        error: payload,
+      });
+    });
+  });
+
+  describe('ridership-over-time selectors', () => {
+    describe('getRidershipOverTimeRequest', () => {
+      it('extends the root selector', () => {
+        const expectation = { one: 'two', three: 4 };
+
+        expect(selectors.getRidershipOverTimeRequest({
+          ridershipOverTime: expectation,
+        })).to.eql(expectation);
+
+        expect(selectors.getRidershipOverTimeRequest({
+          red: 'herring',
+          transportation: {
+            ridershipOverTime: expectation,
+          },
+        })).to.eql(expectation);
+      });
+    });
+
+    describe('getridershipOverTimeData', () => {
+      it('returns undefined when there is no data', () => {
+        expect(selectors.getRidershipOverTimeData({
+          ridershipOverTime: {
+            no: 'data to be seen',
+          },
+        })).to.be.undefined;
+      });
+
+      it('returns undefined when data has no value for ridershipByYear', () => {
+        expect(selectors.getRidershipOverTimeData({
+          ridershipOverTime: {
+            data: {
+              NotridershipByYear: {},
+            },
+          },
+        })).to.be.undefined;
+      });
+
+      it('returns processed data when data has a value for ridershipByYear', () => {
+        const res = {
+          data: [{
+            year: 2001,
+            weekday_sum_ons: 56500470,
+            weekday_sum_offs: 57186610,
+            saturday_sum_ons: 3797456,
+            saturday_sum_offs: 3853425,
+            sunday_sum_ons: 3710746,
+            sunday_sum_offs: 3762772,
+            num_of_yearly_census: 3,
+            sunday_census: true,
+            saturday_census: true,
+            total_sum_ons: 64008672,
+            total_sum_offs: 64802807,
+          }],
+        };
+        const processedData = [{
+          type: 'Weekday',
+          year: 2001,
+          ons: 217309.5,
+        }, {
+          type: 'Saturday',
+          year: 2001,
+          ons: 73028,
+        }, {
+          type: 'Sunday',
+          year: 2001,
+          ons: 71360.5,
+        }];
```

Comment:
👏 👏 Seeing a concrete usage of a selector helps so much when figuring out what it does.